assimilation_code/modules/assimilation/filter_mod.dopplerfold.f90

```diff
@@ -392,30 +398,32 @@ subroutine filter_main()
 
 call trace_message('Before initializing inflation')
 
-call validate_inflate_options(inf_flavor, inf_damping, inf_initial_from_restart, &
+inflation_flavor = set_inflation_flavor(inf_flavor)
+
+call validate_inflate_options(inflation_flavor, inf_damping, inf_initial_from_restart, &
    inf_sd_initial_from_restart, inf_deterministic, inf_sd_max_change,            &
    do_prior_inflate, do_posterior_inflate, output_inflation, compute_posterior)
 
 ! Initialize the adaptive inflation module
-call adaptive_inflate_init(prior_inflate, inf_flavor(1), inf_initial_from_restart(1), &
-   inf_sd_initial_from_restart(1), output_inflation, inf_deterministic(1),            &
-   inf_initial(1), inf_sd_initial(1), inf_lower_bound(1), inf_upper_bound(1),         &
-   inf_sd_lower_bound(1), inf_sd_max_change(1), state_ens_handle,                     &
+call adaptive_inflate_init(prior_inflate, inflation_flavor(PRIOR), inf_initial_from_restart(PRIOR), &
+   inf_sd_initial_from_restart(PRIOR), output_inflation, inf_deterministic(PRIOR),            &
+   inf_initial(PRIOR), inf_sd_initial(PRIOR), inf_lower_bound(PRIOR), inf_upper_bound(PRIOR),         &
+   inf_sd_lower_bound(PRIOR), inf_sd_max_change(PRIOR), state_ens_handle,                     &
```

Comment:
longer term, the inflation options should move to their own namelist.  chunks of code like this are the argument for why being in the filter namelist isn't the right place for them.